### PR TITLE
LPS-105747 Auto changelogs

### PR DESCRIPTION
**Pull request by petershin**
@ling-alan-huang

### CODE REVIEW

**Comment by lawrence-lee** [comment]
Just started reviewing :)

File: build-buildscript.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
+dependencies {
+	classpath group: "com.liferay", name: "com.liferay.gradle.plugins.defaults", version: "6.1.194"
+	classpath group: "com.liferay", name: "com.liferay.gradle.plugins.maven.plugin.builder", version: "1.2.6"
+	classpath group: "de.undercouch", name: "gradle-download-task", version: "3.3.0"
+	classpath group: "gradle.plugin.org.ysb33r.gradle", name: "gradletest", version: "2.0"
+	classpath group: "xalan", name: "xalan", version: "2.7.2"
+
+	if (Boolean.getBoolean("gradle.build.scan.enabled")) {
+		classpath group: "com.gradle", name: "gradle-enterprise-gradle-plugin", version: "3.1.1"
```
Comment (lawrence-lee) [inline build-buildscript.gradle:9]: @ling-alan-huang Why are we adding build scan here? 

We already have build scans enabled https://github.com/liferay/liferay-blade-cli/blob/master/build.gradle#L5

File: changelogs-sdk.gradle [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+							!commitMessage.startsWith("Semver")) {
+
+							commits = commits + "\n" + _format("- [" + gitLogAttributes["TICKET-NUMBER"] + "]: " + commitMessage + " (" + sha.substring(0, 10) + ")")
+
+							ticketNumbers.add(gitLogAttributes["TICKET-NUMBER"])
+						}
+					}
+				}
+
+				String dependencies = ""
```
Comment (lawrence-lee) [inline changelogs-sdk.gradle]: @petershin what's the difference between dependencies and commits?

File: changelogs-sdk.gradle [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+					return Integer.valueOf(ticketNumber1) - Integer.valueOf(ticketNumber2)
+				}
+
+				return ticketType1.compareTo(ticketType2)
+			}
+
+			tickets.each {
+				String ticket ->
+
+				text += "\n[" + ticket + "](https://issues.liferay.com/browse/" + ticket + ")"
```
Comment (lawrence-lee) [inline changelogs-sdk.gradle:248]: I'm not sure why the links don't work, but this is the markdown syntax that works for me. 

`text += "\n[" + ticket + "]: https://issues.liferay.com/browse/" + ticket `

**Comment by lawrence-lee** [comment]
How can I generate the changelogs going forward?

File: changelogs-sdk.gradle (hunk 3 above)
Comment (ling-alan-huang) [reply changelogs-sdk.gradle:248]: I think that is the definition of the link that can be referred by other links.
like:
```
[1]: http://google.com/        "Google"
[2]: http://search.yahoo.com/  "Yahoo Search"
[3]: http://search.msn.com/    "MSN Search"

[Link] [2]
```

**Comment by ling-alan-huang** [comment]
> How can I generate the changelogs going forward?

@lawrence-lee 
`./gradlew -b changelogs-sdk.gradle buildChangelogs`

File: build-buildscript.gradle (hunk 1 above)
Comment (ling-alan-huang) [reply build-buildscript.gradle:9]: @lawrence-lee, this is the first runnable version, I am looking into it and will remove something we don't need.

**Comment by ling-alan-huang** [comment]
@lawrence-lee, I just made some changes and resent: https://github.com/lawrence-lee/liferay-blade-cli/pull/31

**Comment by ling-alan-huang** [comment]
ci:close

**Comment by lawrence-lee** [comment]
@ling-alan-huang, Thanks! 

This is wonderful. I really appreciate all the work you put into making this work!

**Pull request closed without merging**